src/nvim/memfile.c

```diff
@@ -525,82 +472,6 @@ static void mf_rem_used(memfile_T *mfp, bhdr_T *hp)
     mfp->mf_used_first = hp->bh_next;
   else
     hp->bh_prev->bh_next = hp->bh_next;
-
-  mfp->mf_used_count -= hp->bh_page_count;
-  total_mem_used -= hp->bh_page_count * mfp->mf_page_size;
-}
-
-/// Try to release the least recently used block from the used list if the
-/// number of used memory blocks gets too big.
-///
-/// @return  The block header, when release needed and possible.
-///              Resulting block header includes memory block, so it can be
-///              reused.  Page count is checked to be right.
-///          NULL, when release not needed, or not possible.
-///              Not needed when number of blocks less than allowed maximum and
-///              total memory used below 'maxmemtot'.
-///              Not possible when:
-///              - Called while closing file.
-///              - Tried to create swap file but couldn't.
-///              - All blocks are locked.
-///              - Unlocked dirty block found, but flush failed.
-static bhdr_T *mf_release(memfile_T *mfp, unsigned page_count)
-{
-  // don't release while in mf_close_file()
-  if (mf_dont_release)
```

Comment:
@justinmk I will leave the body of `if (getlines) ` because it indicates that when closing we need to get everything from the file to memory. Once we `mmap` files this can be used to use memory unmapped to a file (the swap file).